crates/graphql/src/schema/objects/wallet.rs

```diff
@@ -1,13 +1,65 @@
-use objects::listing::Bid;
+use indexer_core::db::queries;
+use objects::{auction_house::AuctionHouse, listing::Bid};
 use tables::bids;
 
-use super::prelude::*;
+use super::{nft::NftCreator, prelude::*};
+use crate::schema::scalars::PublicKey;
 
 #[derive(Debug, Clone)]
 pub struct Wallet {
     pub address: String,
 }
 
+#[derive(Debug, Clone)]
+pub struct WalletNftCount {
+    wallet: String,
+    creators: Vec<PublicKey<NftCreator>>,
+}
+
+impl WalletNftCount {
+    #[must_use]
+    pub fn new(wallet: String, creators: Vec<PublicKey<NftCreator>>) -> Self {
+        Self { wallet, creators }
+    }
+}
+
+#[graphql_object(Context = AppContext)]
+impl WalletNftCount {
+    fn owned(&self, context: &AppContext) -> FieldResult<i32> {
+        let conn = context.db_pool.get()?;
+        let creators: Vec<String> = self
+            .creators
+            .clone()
+            .into_iter()
+            .map(Into::into)
+            .collect::<Vec<_>>();
+        let count = queries::nft_count::owned(&conn, self.wallet.clone().into(), creators)?;
+        Ok(count.try_into()?)
+    }
+
+    #[graphql(arguments(auction_houses(description = "auction houses to scope wallet counts")))]
+    fn offered(
+        &self,
+        context: &AppContext,
+        auction_houses: Vec<PublicKey<AuctionHouse>>,
+    ) -> FieldResult<i32> {
+        let conn = context.db_pool.get()?;
+        let creators: Vec<String> = self.creators.clone().into_iter().map(Into::into).collect();
+        let auction_houses: Vec<String> = auction_houses.into_iter().map(Into::into).collect();
+        let count = queries::nft_count::offered(
+            &conn,
+            self.wallet.clone().into(),
+            creators,
+            auction_houses,
+        )?;
+        Ok(count.try_into()?)
+    }
+
+    fn listed(&self, _context: &AppContext) -> FieldResult<i32> {
```

Comment:
can you implement listed too. It should take auction_houses as optional argument.

Reply:
@kespinola Do we need to pass auction_houses as argument here?

Reply:
>  It should take auction_houses as optional argument.